### ISSUE TITLE
Task manager reports 0 errors when fails to finish task and logs error

### ISSUE THREAD

**Issue by JoeWaddell** [RuminantFarmSystems/RuFaS#1873]
## Expected Behavior

When an uncaught exception is raised, and the simulation is forced to stop, it should say as much in the command line (as it did previously).

## Current Behavior

While developing https://github.com/RuminantFarmSystems/MASM/pull/1872, noticed that the below message is contradictory. The error files are correctly generated, but the simulation is not reported in the terminal as having failed...after reporting that it did in fact fail. 

![image](https://github.com/user-attachments/assets/685cfc57-9484-4543-8c84-28c7e92952ee)

## Steps to Reproduce

The following feed file should elicit the error on that branch.
```{
  "calf_feeds": [
    202,
    216
  ],
  "growing_feeds": [
    44,
    51,
    95,
    108,
    301,
    305
  ],
  "lac_cow_feeds": [44,
    51,
    95,
    100,
    108,
    301,
    305],
  "close_up_feeds": [
    44,
    51,
    95,
    100,
    108,
    301,
    305
  ],
  "purchased_feeds": [
    {
      "purchased_feed": 202,
      "purchased_feed_cost": 0.1
    },
    {
      "purchased_feed": 216,
      "purchased_feed_cost": 0.1
    },
    {
      "purchased_feed": 44,
      "purchased_feed_cost": 0.1
    },
    {
      "purchased_feed": 51,
      "purchased_feed_cost": 0.1
    },
    {
      "purchased_feed": 95,
      "purchased_feed_cost": 0.1
    },
    {
      "purchased_feed": 108,
      "purchased_feed_cost": 0.1
    },
    {
      "purchased_feed": 301,
      "purchased_feed_cost": 0.1
    },
    {
      "purchased_feed": 305,
      "purchased_feed_cost": 0.1
    },
    {
      "purchased_feed": 100,
      "purchased_feed_cost": 0.1
    }
  ],
  "farm_grown_feeds": [],
  "storage_options": [
    {
      "storage_type": "Bunker Silo",
      "moisture": "Direct Cut",
      "additive": "preservative",
      "packing_density": 200,
      "inoculation": "heterofermentative",
      "bunk_type": "open_floor",
      "ventilation": true,
      "removal_rate": 6,
      "initial_dry_matter": 0
    },
    {
      "storage_type": "Bunker Silo",
      "moisture": "Direct Cut",
      "additive": "preservative",
      "packing_density": 200,
      "inoculation": "heterofermentative",
      "bunk_type": "open_floor",
      "ventilation": true,
      "removal_rate": 6,
      "initial_dry_matter": 0
    }
  ],
  "user_defined_ration_percentages": {
    "calf": [
      {
        "feed_type": 202,
        "ration_percentage": 50
      },
      {
        "feed_type": 216,
        "ration_percentage": 50
      }
    ],
    "growing": [
      {
        "feed_type": 44,
        "ration_percentage": 2.736318407960199
      },
      {
        "feed_type": 51,
        "ration_percentage": 17.226368159203982
      },
      {
        "feed_type": 95,
        "ration_percentage": 26.11940298507463
      },
      {
        "feed_type": 108,
        "ration_percentage": 31.343283582089555
      },
      {
        "feed_type": 301,
        "ration_percentage": 2.3631840796019903
      },
      {
        "feed_type": 305,
        "ration_percentage": 20.211442786069654
      }
    ],
    "close_up": [
      {
        "feed_type": 44,
        "ration_percentage": 2.7275947573503365
      },
      {
        "feed_type": 51,
        "ration_percentage": 20.545518951470065
      },
      {
        "feed_type": 95,
        "ration_percentage": 18.951470067304285
      },
      {
        "feed_type": 100,
        "ration_percentage": 18.986893375841305
      },
      {
        "feed_type": 108,
        "ration_percentage": 18.34927382217499
      },
      {
        "feed_type": 301,
        "ration_percentage": 1.9482819695359546
      },
      {
        "feed_type": 305,
        "ration_percentage": 18.49096705632306
      }
    ],
    "lac_cow": [],
    "tolerance": 0.1,
    "milk_reduction_maximum": 0.5
  }
}
```

**Comment by Isgotkowitz** [comment]
Grooming session: Sister issue to #1856. This issue might be closed by PR that closes that issue. Tagging @matthew7838

**Comment by matthew7838** [comment]
After investigation, indeed this issue is closed by #1856.
